enterprise/frontend/src/embedding-sdk/components/private/SimpleDataPicker/SimpleDataPicker.tsx

```diff
@@ -0,0 +1,101 @@
+import { useDisclosure } from "@mantine/hooks";
+import { type ReactNode, useMemo } from "react";
+
+import { useSearchQuery } from "metabase/api";
+import { isModel } from "metabase/browse/models/utils";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import { Box, Popover } from "metabase/ui";
+import { getQuestionVirtualTableId } from "metabase-lib/v1/metadata/utils/saved-questions";
+import type { SearchResult, TableId } from "metabase-types/api";
+import { SortDirection } from "metabase-types/api/sorting";
+
+import { SimpleDataPickerView } from "./SimpleDataPickerView";
+
+interface SimpleDataPickerProps {
+  filterByDatabaseId: number | null;
+  selectedEntity?: TableId;
+  isInitiallyOpen: boolean;
+  triggerElement: ReactNode;
+  setSourceTableFn: (tableId: TableId) => void;
+}
+
+export function SimpleDataPicker({
+  filterByDatabaseId,
+  selectedEntity,
+  isInitiallyOpen,
+  setSourceTableFn,
+  triggerElement,
+}: SimpleDataPickerProps) {
+  const [isDataPickerOpened, { toggle, close }] =
+    useDisclosure(isInitiallyOpen);
+  const { data, isLoading, error } = useSearchQuery({
+    table_db_id: filterByDatabaseId ? filterByDatabaseId : undefined,
+    models: ["dataset", "table"],
+  });
+
+  const options = useMemo(() => {
+    if (!data) {
+      return [];
+    }
+
+    return sortEntities(
+      data.data.map(entity => {
+        return {
+          ...entity,
+          id: isModel(entity)
```

Comment:
Let's just use `entity.model === "dataset"` (you already pass `"dataset"` a few lines above) and not import some utility from an unrelated module

Reply:
right. I don't know why I decided to import it from `browse`, that definitely looks like a bad abstraction.

Fixed in 0ed61ebf397f787c391445a211d1d9e99fbf4c7c